cesium/open.go

```diff
@@ -97,6 +105,18 @@ func (db *DB) openVirtualOrUnary(ch Channel) error {
 		if isOpen {
 			return nil
 		}
+		if ch.Version != version.CurrentVersion {
+			err = version.Migrate(fs, ch.Version, version.CurrentVersion)
```

Comment:
My gut is that this logic should be inside of `unary` no? I don't see why it needs to be at the cesium level

Reply:
This is kind of weird: the reason it's here is because we must encode a new meta after the migration is done, which requires the `metaECD`: we currently don't store the `metaECD` in any lower level DBs.

Reply:
Added a method in unary that takes `ecd` as an argument to rewrite meta